src/QsCompiler/CSharpGeneration/EntryPoint.fs

```diff
@@ -399,6 +399,11 @@ let private driverSettings context =
         |> (fun (_, value) -> if value = null then "" else value)
         |> literal
 
+    let defaultTargetCapability =
+        context.assemblyConstants.TryGetValue AssemblyConstants.TargetCapability
+        |> (fun (_, value) -> if value = null then "" else value)
+        |> literal
```

Comment:
nit: I think it would make sense to refactor `defaultSimulator`, `defaultExecutionTarget` and `defaultTargetCapability` into one function that gets an assembly constant with a fallback value.

Reply:
```diff
-    let defaultSimulator =
-        context.assemblyConstants.TryGetValue AssemblyConstants.DefaultSimulator
-        |> fun (_, value) -> if String.IsNullOrWhiteSpace value then AssemblyConstants.QuantumSimulator else value
-    let defaultExecutionTarget =
-        context.assemblyConstants.TryGetValue AssemblyConstants.ExecutionTarget
-        |> (fun (_, value) -> if value = null then "" else value)
-        |> literal
-
-    let defaultTargetCapability =
-        context.assemblyConstants.TryGetValue AssemblyConstants.TargetCapability
-        |> (fun (_, value) -> if value = null then "" else value)
-        |> literal
---
+    let CommonFunc(propertyName, defaultValue) = 
+        context.assemblyConstants.TryGetValue propertyName
+        |> (fun (_, value) -> if String.IsNullOrWhiteSpace value then defaultValue else value)
+        |> literal
+
+    let defaultSimulator =
+        CommonFunc(AssemblyConstants.DefaultSimulator, AssemblyConstants.QuantumSimulator)
+
+    let defaultExecutionTarget =
+        CommonFunc(AssemblyConstants.ExecutionTarget, "")
+
+    let defaultTargetCapability =
+        CommonFunc(AssemblyConstants.TargetCapability, "")
```

Reply:
Yes, basically. Some minor F# style nits

- `let` bindings are `camelCase`.
- We prefer curried functions.
- Generally, parameters are ordered from least-likely-to-vary to most-likely-to-vary. The last parameter is usually the most specific or "important" (this supports pipelining).

I would write it something like:

```fsharp
let assemblyConstant defaultValue name =
    ...
```

Reply:
Do you mean this?
```diff
-    let defaultSimulator =
-        context.assemblyConstants.TryGetValue AssemblyConstants.DefaultSimulator
-        |> fun (_, value) -> if String.IsNullOrWhiteSpace value then AssemblyConstants.QuantumSimulator else value
-    let defaultExecutionTarget =
-        context.assemblyConstants.TryGetValue AssemblyConstants.ExecutionTarget
-        |> (fun (_, value) -> if value = null then "" else value)
-        |> literal
-
-    let defaultTargetCapability =
-        context.assemblyConstants.TryGetValue AssemblyConstants.TargetCapability
-        |> (fun (_, value) -> if value = null then "" else value)
-        |> literal
---
+    let assemblyConstant defaultValue name =
+        context.assemblyConstants.TryGetValue name
+        |> (fun (_, value) -> if String.IsNullOrWhiteSpace value then defaultValue else value)
+        |> literal
+
+    let defaultSimulator =
+        assemblyConstant AssemblyConstants.QuantumSimulator AssemblyConstants.DefaultSimulator
+
+    let defaultExecutionTarget =
+        assemblyConstant "" AssemblyConstants.ExecutionTarget
+
+    let defaultTargetCapability =
+        assemblyConstant "" AssemblyConstants.TargetCapability
```

Reply:
Yes, that looks great.

Reply:
This change was resulting in an error in the subsequent line 
`namedArg "createDefaultCustomSimulator" <| customSimulatorFactory defaultSimulator`
(pointing to the beginning of `defaultSimulator` and telling something like "string was expected but got ... expression")

I ended up in the following change that compiled successfully. PLease let me know if there is anything wrong.
```diff
diff --git a/src/QsCompiler/CSharpGeneration/EntryPoint.fs b/src/QsCompiler/CSharpGeneration/EntryPoint.fs
index c79b3ed84..0d1286e02 100644
--- a/src/QsCompiler/CSharpGeneration/EntryPoint.fs
+++ b/src/QsCompiler/CSharpGeneration/EntryPoint.fs
@@ -390,13 +390,19 @@ let private driverSettings context =
         ]
         |> immutableList
 
+    let assemblyConstant defaultValue name =
+        context.assemblyConstants.TryGetValue name
+        |> (fun (_, value) -> if String.IsNullOrWhiteSpace value then defaultValue else value)
+
     let defaultSimulator =
-        context.assemblyConstants.TryGetValue AssemblyConstants.DefaultSimulator
-        |> fun (_, value) -> if String.IsNullOrWhiteSpace value then AssemblyConstants.QuantumSimulator else value
+        assemblyConstant AssemblyConstants.QuantumSimulator AssemblyConstants.DefaultSimulator
 
     let defaultExecutionTarget =
-        context.assemblyConstants.TryGetValue AssemblyConstants.ExecutionTarget
-        |> (fun (_, value) -> if value = null then "" else value)
+        assemblyConstant "" AssemblyConstants.ExecutionTarget
+        |> literal
+
+    let targetCapability =
+        assemblyConstant "" AssemblyConstants.TargetCapability
         |> literal
 
     [
```

Reply:
Oh, I see, `defaultSimulator` needs to be a raw string rather than a C# string literal node. The new code looks good.